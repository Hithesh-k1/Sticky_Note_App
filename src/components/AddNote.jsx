import React, { useState } from "react";

export const AddNote = (props) => {
  const initialFormState = { id: null, title: "", notes: "" };
  const [note, setNote] = useState(initialFormState);

  const handleInputChange = (event) => {
    const { title, value } = event.target;

    setNote({ ...note, [title]: value });
  };

  return (
    <form
      onSubmit={(event) => {
        event.preventDefault();
        // if (!note.title || !note.notes) return;

        props.addNote(note);
        setNote(initialFormState);
      }}
    >
      <div key={note.id}>
        <center>
          {/* <label>Title</label> */}

          {/* <textarea
            className="form-control"
            type="text"
            title="title"
            value={note.title}
            onChange={handleInputChange}
          /> */}
          <div class="form-floating">
            <textarea
              id="floatingTextarea"
              placeholder="Title.."
              className="form-control"
              type="text"
              title="title"
              value={note.title}
              onChange={handleInputChange}
            ></textarea>
            {/* <label for="floatingTextarea">Comments</label> */}
          </div>
        </center>
        {/* <label>Notes</label> */}

        {/* <textarea
          style={{ height: "60%" }}
          className="form-control"
          type="text"
          title="notes"
          value={note.notes}
          onChange={handleInputChange}
        /> */}

<div class="form-floating">
            <textarea
              id="floatingTextarea"
              className="form-control"
              placeholder="Notes.."

          type="text"
          title="notes"
          value={note.notes}
          onChange={handleInputChange}
            ></textarea>
            {/* <label for="floatingTextarea">Comments</label> */}
          </div>
        <button className="btn btn-success">Add Note</button>
      </div>
      <div class="textarea-group mb-3"></div>
    </form>
  );
};

export default AddNote;
