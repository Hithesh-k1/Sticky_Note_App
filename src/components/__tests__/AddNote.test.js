import React from "react";
import { shallow } from "enzyme";
import { AddNote } from "../AddNote";

// let wrappedComponent = shallow(<AddNote />);
const setUp = (props = {}) => {
  const wrappedComponent = shallow(<AddNote {...props} />);
  return wrappedComponent;
};

describe("AddNote component", () => {
  let wrappedComponent;
  let props;

  beforeEach(() => {
    props = {
      addNote: jest.fn(),
      setNote: jest.fn(),
    };

    wrappedComponent = setUp(props);
  });

  it("should match snapshot of AddNote Component", () => {
    expect(wrappedComponent).toMatchSnapshot();
  });

  it("should handle updateNote event", () => {
    let preventDefault = jest.fn();
    const event = { preventDefault };
    wrappedComponent.find("form").props().onSubmit(event);
    expect(props.addNote).toHaveBeenCalled();
  });

  it("should handleInputChange event", () => {
    const event = {
      target: {
        title: "test_title",
        value: "test_value",
      },
    };
    wrappedComponent.find("textarea").at(0).props().onChange(event);
    // expect(props.setNote).toHaveBeenCalled();
  });
});
