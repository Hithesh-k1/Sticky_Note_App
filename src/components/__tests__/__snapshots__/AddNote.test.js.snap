// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNote component should match snapshot of AddNote Component 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    key="null"
  >
    <center>
      <div
        class="form-floating"
      >
        <textarea
          className="form-control"
          id="floatingTextarea"
          onChange={[Function]}
          placeholder="Title.."
          title="title"
          type="text"
          value=""
        />
      </div>
    </center>
    <div
      class="form-floating"
    >
      <textarea
        className="form-control"
        id="floatingTextarea"
        onChange={[Function]}
        placeholder="Notes.."
        title="notes"
        type="text"
        value=""
      />
    </div>
    <button
      className="btn btn-success"
    >
      Add Note
    </button>
  </div>
  <div
    class="textarea-group mb-3"
  />
</form>
`;
